source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp

```diff
@@ -128,180 +127,119 @@ MPxNode::SchedulingType MaterialXNode::schedulingType() const
 	return MPxNode::SchedulingType::kParallel;
 }
 
-bool MaterialXNode::setInternalValue(const MPlug &plug, const MDataHandle &dataHandle)
+bool MaterialXNode::getInternalValue(const MPlug& plug, MDataHandle& dataHandle)
 {
-	if (!materialXData) return false;
+    if (plug == DOCUMENT_ATTRIBUTE)
+    {
+        dataHandle.set(_documentFilePath);
+    }
+    else if (plug == ELEMENT_ATTRIBUTE)
+    {
+        dataHandle.set(_xmlElementPath);
+    }
+    else
+    {
+        return MPxNode::getInternalValue(plug, dataHandle);
+    }
 
-	if (plug == DOCUMENT_ATTRIBUTE)
-	{
-		//MString documentContent = dataHandle.asString();
-		//materialXData->createDocument(documentContent.asChar());
-	}
-	else if (plug == ELEMENT_ATTRIBUTE)
-	{
-		if (materialXData->getDocument())
-		{
-			//MString elementPath = dataHandle.asString();
-			//materialXData->element = materialXData->doc->getDescendant(elementPath.asChar());
-		}
-	}
-	else if (plug == _outAttr)
-	{
-		return MPxNode::setInternalValue(plug, dataHandle);
-	}
-	else
-	{
-		std::string plugName = plug.partialName().asChar();
-		if (plug.isChild())
-		{
-			MPlug plug2 = plug.parent();
-			plugName = plug2.partialName().asChar();
-		}
-		if (_attributeElementPairMap.count(plugName) > 0)
```

Comment:
Removing the dynamically created shader input attributes for now. Let's get the foundation right first and return to these later.

Reply:
This should be removed permanently. Element inputs should never be exposed as DG attributes.